docs/proposals/block-storage-time-series-deletion.md

```diff
@@ -100,7 +100,7 @@ Similar to the chunk storage deletion implementation, the initial filtering of t
 #### Storing tombstones in object store 
 
 
-The Purger will store the tombstone entries in a separate folder called `tombstones` in the object store (e.g. S3 bucket) in the respective tenant folder. Each tombstone can have a separate JSON file outlining all the necessary information about the deletion request such as the parameters passed in the request, as well as some meta-data such as the creation date of the request.  The name of the file can be a hash of the API parameters (start, end, markers). This way if a user calls the API twice by accident with the same parameters, it will only create one tombstone. To keep track of the request state, filename extensions can be used. This will allow the tombstone files to be immutable. The 4 different file extensions will be `received, deleting, syncing, processed`.
+The Purger will write the new tombstone entries in a separate folder called `tombstones` in the object store (e.g. S3 bucket) in the respective tenant folder. Each tombstone can have a separate JSON file outlining all the necessary information about the deletion request such as the parameters passed in the request, as well as some meta-data such as the creation date of the file.  The name of the file can be a hash of the API parameters (start, end, markers). This way if a user calls the API twice by accident with the same parameters, it will only create one tombstone. To keep track of the request state, filename extensions can be used. This will allow the tombstone files to be immutable. The 4 different file extensions will be `received, deleting, syncing, processed`. Each time the deletion request moves to a new state, a new file will be added with the same content but a different extension to indicate the new state. The file containing the previous state will be deleted once the new one is created. 
```

Comment:
Few comments:
1. I would suggest to rename `received` state into `pending` (because it's pending waiting to be processed by the compactor).
2. I'm not sure we actually need `syncing`. Let's say we remove it. The tombstone in `processed` state contains the timestamp of when it was processed, so the querier can check that timestamp and stop enforcing the timestamp after "timestamp + delay period".
3. "The file containing the previous state will be deleted once the new one is created": we also have to address the case the creation of the new state succeed and the deletion fails. Since state changes can only flow in one direction (received -> deleting -> syncing -> processed), if the next compactor run will find multiple states for the same tombstone then it can clean up "lower" states.

Reply:
Thank you for the suggestions. I've updated the proposal to reflect that.